opsgenie/resource_opsgenie_incident_template_test.go

```diff
@@ -0,0 +1,152 @@
+		Name: "opsgenie_incident_template",
+		F:    testSweepIncidentTemplate,
+	})
+}
+
+func testSweepIncidentTemplate(region string) error {
+	meta, err := sharedConfigForRegion()
+	if err != nil {
+		return err
+	}
+	client, err := incident.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	result, err := client.GetIncidentTemplate(context.Background(), &incident.GetIncidentTemplateRequest{})
+	if err != nil {
+		return err
+	}
+	if result != nil {
+		for _, value := range result.IncidentTemplates["incidentTemplates"] {
+			if strings.HasPrefix(value.Name, "Incident Template Name-") {
+				log.Printf("Destroying incident template %s", value.Name)
+				deleteRequest := incident.DeleteIncidentTemplateRequest{IncidentTemplateId: value.IncidentTemplateId}
+				if _, err := client.DeleteIncidentTemplate(context.Background(), &deleteRequest); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func TestAccOpsGenieIncidentTemplate_basic(t *testing.T) {
+	config := testAccOpsGenieIncidentTemplate_basic(acctest.RandString(6))
+	resource.Test(t, resource.TestCase{
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckOpsGenieIncidentTemplateDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckOpsGenieIncidentTemplateExists(),
+				),
+			},
+		},
+	})
+}
+
+func testCheckOpsGenieIncidentTemplateDestroy(s *terraform.State) error {
+	client, err := incident.NewClient(testAccProvider.Meta().(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "opsgenie_incident_template" {
+			continue
+		}
+		result, err := client.GetIncidentTemplate(context.Background(), &incident.GetIncidentTemplateRequest{})
+		if err != nil {
+			if result != nil {
+				for _, value := range result.IncidentTemplates["incidentTemplates"] {
+					if strings.HasPrefix(value.Name, "Incident Template Name-") {
+						x := err.(*ogClient.ApiError)
+						if x.StatusCode != 404 {
+							return errors.New(fmt.Sprintf("Incident template still exists: %s", x.Error()))
+						}
+					}
+				}
+			}
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+func testCheckOpsGenieIncidentTemplateExists() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client, err := incident.NewClient(testAccProvider.Meta().(*OpsgenieClient).client.Config)
+		if err != nil {
+			return err
+		}
+		result, err := client.GetIncidentTemplate(context.Background(), &incident.GetIncidentTemplateRequest{})
+		if err != nil {
+			if result != nil {
+				for _, value := range result.IncidentTemplates["incidentTemplates"] {
+					if strings.HasPrefix(value.Name, "Incident Template Name-") {
+						log.Printf("Incident template found.")
+					} else {
+						return fmt.Errorf("incident template does not exist (and it should)")
+					}
```

Comment:
i dont understand here 
i we found we should exit the for loop and else should exist out of the loop

Reply:
in this case we never iterate the second element ?

Reply:
The logic here is.. we iterate through the loop and if we don't find the template then we inform the user it should exists but it didn't. Changes are done to logic here. please do have a look.